lua/neogit/lib/popup/builder.lua

```diff
@@ -31,46 +33,163 @@ function M:env(x)
   return self
 end
 
-function M:new_action_group()
-  table.insert(self.state.actions, {})
+function M:new_action_group(heading)
+  table.insert(self.state.actions, { { heading = heading or "" } })
   return self
 end
 
---@param parse Whether the switch is internal to neogit or should be included in the cli command.
---             If `false` we don't include it in the cli comand.
-function M:switch(key, cli, description, enabled, parse)
-  if enabled == nil then
-    enabled = false
+function M:new_action_group_if(cond, heading)
+  if cond then
+    return self:new_action_group(heading)
   end
 
-  if parse == nil then
-    parse = true
+  return self
+end
+
+function M:group_heading(heading)
+  table.insert(self.state.actions[#self.state.actions], { heading = heading })
+  return self
+end
+
+function M:group_heading_if(cond, heading)
+  if cond then
+    return self:group_heading(heading)
   end
 
-  table.insert(self.state.switches, {
-    id = "-" .. key,
+  return self
+end
+
+---@param opts.internal boolean Whether the switch is internal to neogit or should be included in the cli command.
+--                              If `true` we don't include it in the cli comand.
```

Comment:
Perhaps we should add incompatible here as well,

or in general maybe document the popup a little more?

Reply:
I'm trying to just add things on an as-needed basis (for the most part, at least). Haven't found a place where two options need to be exclusive yet.

I can add more documentation, thats a good idea.